ops/model.py

```diff
@@ -985,6 +1006,52 @@ def location(self):
         return self._location
 
 
+class Container:
+    """Represents a named (Kubernetes) container in an application.
+
+    Attributes:
+        name: The name of the container from metadata.yaml (eg, 'postgres').
+    """
+
+    def __init__(self, name):
+        self.name = name
+
+        socket_path = '/charm/containers/{}/pebble/.pebble.socket'.format(name)
+        self._pebble = pebble.API(socket_path=socket_path)
+
+    @property
+    def pebble(self) -> 'pebble.API':
+        """Return the low-level Pebble API instance for this container."""
+        return self._pebble
+
+    def autostart(self):
+        """Autostart all default services."""
+        self._pebble.autostart_services()
+
+    def start(self, *service_names: typing.List[str]):
+        """Start given service(s) by name."""
+        self._pebble.start_services(service_names)
+
+    def stop(self, *service_names: typing.List[str]):
+        """Stop given service(s) by name."""
+        self._pebble.stop_services(service_names)
+
+    def add_layer(self, config: typing.Union[str, typing.Dict, 'pebble.Config']):
+        """Dynamically add a service (Pebble) configuration layer.
+
+        Args:
+            config: A YAML string, config dict, or pebble.Config object
+                containing the Pebble configuration layer to add
+        """
+        if isinstance(config, (str, dict)):
+            config = pebble.Config(config)
+        self._pebble.add_layer(config)
+
+    def get_config(self) -> 'pebble.Config':
```

Comment:
Related to above.

Reply:
This one's a bit trickier, because I think you do want the higher-level `pebble.Setup` object returned here. However, I've made the low-level `PebbleClient.get_config()` function return the raw string, so if a developer really needs that level of control that can dip down to the `unit.pebble.get_config()` function.